contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -0,0 +1,875 @@
+	// use ListWatch watching pods using the client by default
+	return k.NewPluginConfig(terminate, mux, createAllPodsLW(k.client))
+}
+
+func (k *KubernetesScheduler) NewPluginConfig(terminate <-chan struct{}, mux *http.ServeMux,
+	podsWatcher *cache.ListWatch) *PluginConfig {
+
+	// Watch and queue pods that need scheduling.
+	updates := make(chan queue.Entry, k.schedcfg.UpdatesBacklog)
+	podUpdates := &podStoreAdapter{queue.NewHistorical(updates)}
+	reflector := cache.NewReflector(podsWatcher, &api.Pod{}, podUpdates, 0)
+
+	// lock that guards critial sections that involve transferring pods from
+	// the store (cache) to the scheduling queue; its purpose is to maintain
+	// an ordering (vs interleaving) of operations that's easier to reason about.
+	kapi := &k8smScheduler{internal: k}
+	q := newQueuer(podUpdates)
+	podDeleter := &deleter{
+		api: kapi,
+		qr:  q,
+	}
+	eh := &errorHandler{
+		api:     kapi,
+		backoff: backoff.New(k.schedcfg.InitialPodBackoff.Duration, k.schedcfg.MaxPodBackoff.Duration),
+		qr:      q,
+	}
+	startLatch := make(chan struct{})
+	eventBroadcaster := record.NewBroadcaster()
+	runtime.On(startLatch, func() {
+		eventBroadcaster.StartRecordingToSink(k.client.Events(""))
+		reflector.Run() // TODO(jdef) should listen for termination
+		podDeleter.Run(updates, terminate)
+		q.Run(terminate)
+
+		q.installDebugHandlers(mux)
+		podtask.InstallDebugHandlers(k.taskRegistry, mux)
+	})
+	return &PluginConfig{
+		Config: &plugin.Config{
+			MinionLister: nil,
+			Algorithm: &kubeScheduler{
+				api:        kapi,
+				podUpdates: podUpdates,
+			},
+			Binder:   &binder{api: kapi},
+			NextPod:  q.yield,
+			Error:    eh.handleSchedulingError,
+			Recorder: eventBroadcaster.NewRecorder(api.EventSource{Component: "scheduler"}),
+		},
+		api:      kapi,
+		client:   k.client,
+		qr:       q,
+		deleter:  podDeleter,
+		starting: startLatch,
+	}
+}
+
+type PluginConfig struct {
+	*plugin.Config
+	api      schedulerInterface
+	client   *client.Client
+	qr       *queuer
+	deleter  *deleter
+	starting chan struct{} // startup latch
+}
+
+func NewPlugin(c *PluginConfig) PluginInterface {
+	return &schedulingPlugin{
+		config:   c.Config,
+		api:      c.api,
+		client:   c.client,
+		qr:       c.qr,
+		deleter:  c.deleter,
+		starting: c.starting,
+	}
+}
+
+type schedulingPlugin struct {
+	config   *plugin.Config
+	api      schedulerInterface
+	client   *client.Client
+	qr       *queuer
+	deleter  *deleter
+	starting chan struct{}
+}
+
+func (s *schedulingPlugin) Run(done <-chan struct{}) {
+	defer close(s.starting)
+	go runtime.Until(s.scheduleOne, pluginRecoveryDelay, done)
+}
+
+// hacked from GoogleCloudPlatform/kubernetes/plugin/pkg/scheduler/scheduler.go,
+// with the Modeler stuff removed since we don't use it because we have mesos.
+func (s *schedulingPlugin) scheduleOne() {
+	pod := s.config.NextPod()
+	log.V(3).Infof("Attempting to schedule: %v", pod)
+	dest, err := s.config.Algorithm.Schedule(pod, s.config.MinionLister)
```

Comment:
It took me a while to find Schedule() defined in
contrib/mesos/pkg/scheduler/types.go
so maybe can you add a comment pointing there?